apps/system/flash_test/flash_test.c

```diff
@@ -59,80 +59,570 @@
+	fill_buffer(buffer);
+	printf("Read size %d at addr 0x%x: \n",size, addr);
+	lldbg("lldb: Flash Read start\n");
 
 	while (remaining > 0) {
+		//sys_start_time = TICK2MSEC(clock_systimer());
+		sys_start_time = get_time();
+		
 		read_size = BUF_SIZE < remaining ? BUF_SIZE : remaining;
-		ret = MTD_READ(dev_mtd, addr + offset, read_size, buffer);
+		ret = MTD_READ(dev_mtd, addr + offset, read_size, (uint8_t *) rd_buffer);
 		if (ret <= 0) {
 			printf("Read Failed : %d\n", ret);
 			free(buffer);
+			free(rd_buffer);
+			return ERROR;
+		}
+		sys_end_time = get_time();
+		//sys_end_time = TICK2MSEC(clock_systimer());
+		systick_time_ms += sys_end_time - sys_start_time;
+		/* Compare the contents */
+		if (memcmp(rd_buffer, buffer, read_size)) {
+			printf("ERROR: Flash Read value comparison failed at %p \n", addr + offset);
+			fflush(stdout);
+			free(buffer);
+			free(rd_buffer);
 			return ERROR;
 		}
-
 		/* Print read buffer */
+#if 0
 		for (buf_idx = 0; buf_idx < ret; buf_idx++) {
 			if ((buf_idx % 16) == 0) {
 				printf("\n");
 			}
-			printf("0x%02x ", buffer[buf_idx]);
+			printf("0x%02x ", (uint8_t *) rd_buffer[buf_idx]);
 		}
+#endif
 		offset += ret;
 		remaining -= ret;
 	}
+	lldbg("lldb: Flash Read End\n");
+	printf("\n Systick time taken %ld(ms)\n", systick_time_ms);
 	printf("\n");
-
 	free(buffer);
+	free(rd_buffer);
 
 	return OK;
 }
 
+static int flash_write(unsigned long addr, int size, flash_t flash)
+{
+	int ret;
+	int offset;
+	int write_size;
+	int remaining;
+	FAR uint32_t *buffer;
+	FAR struct mtd_dev_s *dev_mtd = NULL;
+
+	if (addr == 0 || size <= 0) {
+		printf("Invalid arguments, offset 0x%x, Write size %d\n", addr, size);
+		return ERROR;
+	}
+	if (flash == FLASH_INTERNAL) {
+		dev_mtd = up_flashinitialize();
+		if (!dev_mtd) {
+			printf("up_flashinitialize Failed\n");
+			return ERROR;
+		}
+	}
+#ifdef CONFIG_SECOND_FLASH_PARTITION
+	else if (flash == FLASH_EXT) {
+		struct spi_dev_s *spi = up_spiinitialize(1);
+
+#ifdef CONFIG_MTD_JEDEC
+		dev_mtd = jedec_initialize(spi);
+		if (dev_mtd == NULL) {
+			lldbg("Jedec Init failed\n");
+			return ERROR;
+		}
+#elif defined(CONFIG_MTD_W25)
+		dev_mtd = w25_initialize(spi);
+		if (dev_mtd == NULL) {
+			lldbg("w25 Init failed\n");
+			return ERROR;
+		}
+#endif
+	}
+#endif
+
+	buffer = (FAR uint32_t *)malloc(BUF_SIZE);
+	if (!buffer) {
+		printf("Failed to allocate buffer %d\n", BUF_SIZE);
+		return ERROR;
+	}
+	fill_buffer(buffer);
+
+	offset = 0;
+	remaining = size;
+
+	printf("Write 0x%x: \n", addr);
+	//sys_start_time = TICK2MSEC(clock_systimer());
+	sys_start_time = get_time();
+	lldbg("lldb: Flash Write Start\n");
+	while (remaining > 0) {
+		write_size = BUF_SIZE < remaining ? BUF_SIZE : remaining;
+		ret = MTD_WRITE(dev_mtd, addr + offset, write_size, (const uint8_t *)buffer);
+		if (ret <= 0) {
+			printf("Write Failed : %d\n", ret);
+			free(buffer);
+			return ERROR;
+		}
+		offset += ret;
+		remaining -= ret;
+	}
+	lldbg("lldb: Flash Write End\n");
```

Comment:
Is there any reason to be lldbg, not printf? We should not use lldbg at normal code.

Reply:
nothing specific on using updated it to use printf.